App.js

```diff
@@ -64,23 +83,29 @@ function WithCustomBackBehavior(props) {
   return props.children;
 }
 
+const useFocusedRouteParams = (navigation) => {
+  const state = useNavigationState(state => state)
+  return state.routes?.[state.index]?.params
+}
+
 function LeftHandNav({navigation}) {
+  const focusedRouteParams = useFocusedRouteParams(navigation);
   return (
     <View style={{flex: 1}}>
       <LHNHeader navigation={navigation} />
-      <Pressable style={{flexDirection: 'row', margin: 10, alignItems: 'center'}} onPress={() => navigation.push('Chat', {id: 1})}>
+      <Pressable style={{flexDirection: 'row', margin: 10, alignItems: 'center'}} disabled={focusedRouteParams?.id === 1} onPress={() => navigation.push('Chat', {id: 1})}>
```

Comment:
Thanks for fixing this up. Any idea why `route` doesn't contain the `params` here?

Reply:
It does but it is a route for the LHN. We need to get top chat params

Reply:
Ohhh I see that’s why it was undefined. Make sense to get the top chat’s param here.